### PR TITLE
Additions to camera and aovs

### PR DESCRIPTION
**Pull request by jdent02**

### CODE REVIEW

File: translators/camera.py [hunk 1]
```diff
@@ -368,6 +381,12 @@ def __view_camera_params(self, aspect_ratio):
                       'shift_y': self.__shift_y,
                       'film_dimensions': asr.Vector2f(film_width, film_height)}
 
+        if model == 'fisheyelens_camera':
+            if self.bl_camera.data.appleseed.fisheye_projection_type is not 'none':
+                params['projection_type'] = self.bl_camera.data.appleseed.fisheye_projection_type
+            else:
+                print("Spherical camera not supported for interactive rendering")
```
Comment (dictoon) [inline translators/camera.py]: As you probably noticed, at this point I'm getting very confused :) Why is this message mentioning "spherical camera" when you're in a block dealing with a fisheye lens camera? 

Moreover, all camera models are supported by interactive rendering, at least in appleseed core.

File: translators/scene.py [hunk 2]
```diff
@@ -682,6 +682,8 @@ def __translate_frame(self):
                 aovs.insert(asr.AOV('diffuse_aov', {}))
             if asr_scene_props.direct_diffuse_aov:
                 aovs.insert(asr.AOV('direct_diffuse_aov', {}))
+            if asr_scene_props.screen_space_velocity_aov:
```
Comment (dictoon) [inline translators/scene.py]: Kind of weird to handle `screen space velocity` between `direct diffuse` and `indirect diffuse`. Order alphabetically?

File: properties/nodes/__init__.py [hunk 3]
```diff
@@ -82,21 +82,21 @@ def acknowledge_connection(self, context):
...
         return context.bl_idname == "AppleseedNodeTree" and renderer == 'APPLESEED_RENDER'
 
     def get_node_name(self):
-        """Return the node's name, including appended pointer."""
+        """Return the node's name, including appended pointer"""
         return util.join_names_underscore(self.name, str(self.as_pointer()))
 
     def traverse_tree(self, material_node):
-        """Iterate inputs and traverse the tree backward if any inputs are connected.
+        """Iterate inputs and traverse the tree backward if any inputs are connected
```
Comment (dictoon) [inline properties/nodes/__init__.py]: I'm pretty sure you can (and should) preserve periods in the middle of the description.

File: properties/nodes/__init__.py [hunk 4]
```diff
@@ -106,9 +106,9 @@ def traverse_tree(self, material_node):
 
 
 class AppleseedOSLNodeCategory(NodeCategory):
-    """Node category for extending the Add menu, toolbar panels and search operator.
+    """Node category for extending the Add menu, toolbar panels and search operator
```
Comment (dictoon) [inline properties/nodes/__init__.py]: Here too.

File: properties/camera.py [hunk 5]
```diff
@@ -42,10 +42,20 @@ class AppleseedCameraSettings(bpy.types.PropertyGroup):
     camera_model = bpy.props.EnumProperty(name="camera_model",
                                           items=[('pinhole', "Pinhole", ''),
                                                  ('thinlens', "Thin Lens", ''),
-                                                 ('spherical', "Spherical", '')],
+                                                 ('spherical', "Spherical", ''),
+                                                 ('fisheyelens_camera', "Fisheye", "")],
```
[the remark this answers is not in the document]
Comment (jdent02) [reply properties/camera.py]: Cause that's what it's called in appleseed...

```
namespace
{
    const char* Model = "fisheyelens_camera";

    class FisheyeLensCamera
      : public PerspectiveCamera
    {
```

File: translators/camera.py [hunk 6]
```diff
@@ -73,6 +73,9 @@ def create_entities(self, scene):
 
         model = cam_mapping[self.bl_camera.data.type]
 
+        if model == 'spherical_camera' and not self.bl_camera.data.appleseed.fisheye_projection_type == 'none':
```
[the remark this answers is not in the document]
Comment (jdent02) [reply translators/camera.py]: See long explanation below....

File: translators/camera.py [hunk 7]
```diff
@@ -203,6 +209,15 @@ def __thin_lens_camera_params(self, scene, aspect_ratio, film_width, film_height
 
         return cam_params
 
+    def __fisheye_camera_params(self, scene, aspect_ratio, film_width, film_height):
+        camera = self.bl_camera
+
+        cam_params = self.__pinhole_camera_params(scene, aspect_ratio, film_width, film_height)
```
[the remark this answers is not in the document]
Comment (jdent02) [reply translators/camera.py]: Pinhole camera populates all the 'standard' params like focal length, film size, aspect ratio, etc...

File: translators/camera.py [hunk 8]
```diff
@@ -343,7 +358,8 @@ def __view_camera_params(self, aspect_ratio):
         self.__matrix = self.bl_camera.matrix_world
         cam_mapping = {'PERSP': 'pinhole_camera',
                        'ORTHO': 'orthographic_camera',
-                       'PANO': 'pinhole_camera'}
+                       'PANO': 'fisheyelens_camera'}
```
[the remark this answers is not in the document]
Comment (jdent02) [reply translators/camera.py]: See below

File: translators/camera.py (hunk 1 above)
Comment (jdent02) [reply translators/camera.py]: See below

**Comment by jdent02** [comment]
So the camera issues are tied to Blender itself.  It only has three camera modes: perspective, ortho and panoramic.  In order for the render to match the 3D 'through camera' view, we have to stick to those three.  I can't add my own to that list (it's hard coded in C).  Cycles puts it's fisheye lens categories into the panoramic category, so I've done the same.  I suppose they could be put under the perspective view...

As far as spherical not being supported in interactive mode.. I understand that appleseed can do it but the problem is Blender's 3D viewport.  It allows for parts of the scene outside the camera to be visible in interactive mode.  In Cycles it handles the panoramic camera by repeating the camera view (creating a tiling effect).  I haven't figured out how to do that yet, so for now the spherical mode is disabled in interactive.

File: properties/camera.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (dictoon) [reply properties/camera.py]: My point is about consistency, all cameras use the `_camera` suffix in appleseed, e.g.

    const char* Model = "pinhole_camera";

    class PinholeCamera
      : public PerspectiveCamera

File: translators/camera.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (dictoon) [reply translators/camera.py]: Can we rename `__pinhole_camera_params()` to `__base_camera_params()` or something like that to help clearing any potential confusion?

File: translators/camera.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (dictoon) [reply translators/camera.py]: If that's what Blender users expect and what Cycles does, then I think we're good.

File: properties/camera.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (jdent02) [reply properties/camera.py]: Actually.....  turns out this property isn’t even used 😳. It’s a leftover from an earlier version of the plugin.

**Pull request merged**